### DISCUSSION TITLE
Private tab/container

### DISCUSSION THREAD

**Discussion by mr-cheffy** [zen-browser/desktop#767] [Ideas]
### What feature would you like to see?

It would be nice to have an option to create a new tab in a private container, like for example in the Floorp browser. 
Maybe even go a step further and give users ability to make history separate in each container or give them ability to change when the history should be cleared.

 
 ~ _Created automatically from issue: https://github.com/zen-browser/desktop/issues/681_

**Comment by Nitrousoxide** [comment]
I would like this as well.  I use this feature frequently on Floorp for sites like Twitter/X/Facebook, where I want to be certain that If I do log in that any cookies, et al get blown out after I close the tab.

**Comment by AQDuck97** [comment]
To add to this, the way Floorp did it was modifying the MAC (Multi-Account Containers) extension with a new "Private" container that basically worked as a private mode window and would be wiped when closed.

**Reply by Lu9-ST** [reply]
Basically - and I believe Waterfox does it as well. Probably shouldn't be too difficult to take a look at how they were implemented in both WF and Floorp, to implement it into Zen. 
One curious thing is I once manually moved a Floorp profile into Zen, and it did create a "Private" container, but of course, it just acts like a regular container

**Reply by d5364t54ytfr4** [reply]
Yep, there's also the `Temporary Containers` addon whose developer died long ago. Should work in Zen, though might have bugs. The button shows up in the bottom.
And using `Container Sidebar` helps with managing containers in Firefox.
BTW, I don't think they modify anything. They just delete the container and spin a new one I think.

**Comment by d5364t54ytfr4** [comment]
Take a look at these two related discussions:

https://github.com/zen-browser/desktop/discussions/2078
https://github.com/zen-browser/desktop/discussions/821
